Do not delete key from unused map if field value cannot be set

mback2k opened the pull request:
Track unusable keys in case the value cannot be set. This helps troubleshoot issues in case a struct was unexported. It took me over a hour to find such an issue once, I would like to avoid this in the future.

mitchellh commented:
Hi, its been over a year so I apologize but if you're interested in this I'd love to see a test case added here.

mback2k commented:
Thanks for your response, I just rebased and force-pushed my change, added a new test and updated existing metadata tests to include an unexported field. Please take a look.

mitchellh commented:
Perfect. Thank you!
